tuf/signed/verify.go

```diff
@@ -77,10 +77,11 @@ func VerifySignatures(s *data.Signed, roleData data.BaseRole) error {
 		if key.ID() != sig.KeyID {
 			return ErrInvalidKeyID{}
 		}
-		if err := VerifySignature(msg, sig, key); err != nil {
+		if err := VerifySignature(msg, &sig, key); err != nil {
 			logrus.Debugf("continuing b/c %s", err.Error())
 			continue
 		}
+		s.Signatures[i].IsValid = sig.IsValid
```

Comment:
We could also have `Sign` set the `IsValid` field once it's finished verifying it, too.  Another way to do this, if we want `VerifySignature` to actually do the mutation (if we cared about `Sign` signatures having the correct boolean set):

```
	for i := range s.Signatures {
		sig := &(s.Signatures[i])
                ...
		if err := VerifySignature(msg, sig, key); err != nil {
                    ...
```

So we can skip the additional assignment.